subsystem/claw.py

```diff
@@ -0,0 +1,69 @@
+import math
+import config
+import constants
+import wpilib
+from robotpy_toolkit_7407 import Subsystem
+from robotpy_toolkit_7407.motors.rev_motors import SparkMax
+from robotpy_toolkit_7407.pneumatics.pistons import SingleSolenoidPiston
+
+class Claw(Subsystem):
+    
+    def __init__(self, claw_motor: SparkMax):
+        """
+        Constructor 
+
+        Args:
+            claw_motor (SparkMax): The motor that controls the claw. Pass with the correct
+            can_id and turn/drive config. 
+        """
+        self.claw_motor = claw_motor
+        self.claw_close_piston: SingleSolenoidPiston = SingleSolenoidPiston(1, wpilib.PneumaticsModuleType.REVPH, 4)
+        self.claw_motor_initialized: bool = False
+        self.claw_compressed: bool = False
+
+    def init(self):
+        self.claw_motor.init()
+        self.zero()
+        self.claw_motor_initialized = True
+
+    def set_angle(self, pos: float):
+        """
+        Set the angle of the claw
+
+        Args:
+            pos (float): Angle in radians (0, 2pi)
+        """
+        self.claw_motor.set_target_position(
+            (pos / (math.pi * 2)) * constants.claw_gear_ratio  # 80 rotations per 360 degrees
+        )   
+
+    def get_angle(self):
+        # Return through bore encode value 
+        return self.claw_motor.get_sensor_position() / constants.claw_gear_ratio
```

Comment:
What are the unit here? Does this sensor need any scaling?

Reply:
Radians divided by the gear ratio (gear ratio is a dummy value now as the claw has not been put on the robot yet)

Reply:
I am now passing SparkMax inside of robot_systems. Let me know if that was what you were trying to ask me to implement. I also replaced the piston back to `DoubleSolenoidPiston` as per what @sidharthmrao's told me.